unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Audio/AudioContainer.cs

```diff
@@ -38,14 +33,12 @@ void Awake()
         {
             if (e.clips.Length == 0)
             {
-                Debug.LogWarning("There are no clips in the audio event '" + e.name + "' (" + name + " -> " + identifier + ")");
+                Debug.LogWarning("There are no clips in the audio event '" + e.name + "' (" + name + ")");
                 continue;
             }
 
-            e.Initialize();
-
-            // Set up AudioSource component for event
-            e.source = gameObject.AddComponent(typeof(AudioSource)) as AudioSource;
+            // Add AudioSource component for event
+            e.Initialize(gameObject.AddComponent(typeof(AudioSource)) as AudioSource);
```

Comment:
Maybe we can just pass the `AudioContainer` here, and move all the `AddComponent` and `AudioSource` setup inside that method? This way the entire initialization gets encapsulated inside `AudioEvent`.